src/buffer.cc

```diff
@@ -627,6 +625,37 @@ bool Buffer::is_modified() const
             not m_current_undo_group.empty());
 }
 
+bool Buffer::is_file_modified() const
+{
+    if (not (m_flags & Flags::File))
+        throw runtime_error("Buffer doesn't have a file");
+
+    {
+        MappedFile fd{m_name};
+        const size_t eol_offset = (m_eol_format == EolFormat::Crlf ? 2 : 1);
+
+        if (m_contents_size != fd.st.st_size)
+            return true;
+
+        if (m_byte_order_mark == ByteOrderMark::Utf8)
+            fd.data += 3;
+
+        for (size_t i = 0; i < line_count(); i++)
+        {
+            const StringDataPtr& bufdata_str = line_storage(i);
+            const size_t bufdata_len = bufdata_str->length - 1;
+
+            if ((!bufdata_len and fd.data[eol_offset - 1] != '\n')
```

Comment:
That looks strange, you are looking at the first or second character of `fd.data`, and checking if its `\n`, I think thats not what you meant.

Reply:
I don't know if the behavior of `strncmp()` is well defined when its size argument is 0, so I'm verifying that the corresponding line in the file is empty.

Reply:
`std::equal` will make all that cleaner.